### PR TITLE
Improve typechecking

### PR DESCRIPTION
**Pull request by LMS007**
- new-note-btn.js
- search-status-bar.js
- selection-tabs.js
- annotation-action-bar.js
- annotation-quote.js

### CODE REVIEW

File: src/sidebar/components/new-note-btn.js [hunk 1]
```diff
@@ -21,9 +31,9 @@ function NewNoteButton({ annotationsService, settings }) {
     }
     const annot = {
       target: [],
-      uri: topLevelFrame.uri,
+      uri: topLevelFrame ? topLevelFrame.uri : '',
```
Comment (LMS007) [inline src/sidebar/components/new-note-btn.js:34]: I wonder if this even a valid state when `mainFrame` returns null? but using a blank string seems reasonable.

Comment (robertknight) [reply src/sidebar/components/new-note-btn.js:34]: This component shouldn't be shown when `mainFrame` returns `null`. We definitely can't create an annotation in that situation, so I would suggest to just bail out of the callback in that case, or just make the component render nothing.

File: src/sidebar/components/new-note-btn.js [hunk 2]
```diff
@@ -21,9 +31,9 @@ function NewNoteButton({ annotationsService, settings }) {
...
     };
-    annotationsService.create(annot);
+    annotationsService.create1(annot);
```
Comment (robertknight) [inline src/sidebar/components/new-note-btn.js:36]: This looks like an accidental change.

Comment (LMS007) [reply src/sidebar/components/new-note-btn.js:36]: oops!

**Pull request merged**